API.md

```diff
@@ -0,0 +1,66 @@
+
+Important: call this method at least once, in your `setup()`.
+
+### `void digitalPinWrite(uint8_t pin, bool value, uint8_t device = 0)`
+
+Sets the state ("on" or "off") of one power switch. Function arguments:
+
+- `pin`: the ordinal number of the power switch on the board. It ranges from `0` to `7`. Use an optional value `ALL` to select and set the state of all power switches simultaneously.
+- `value`: selects the state of the power switch. Values "on" and "off" may be put in different data types: `HIGH` or `LOW` / `true` or `false` / `1` or `0`.
+- `device`: the ordinal number of the device on the bus. It ranges from `0` to `n - 1`, where `n` is the last device number. Use an optional value `ALL` to select and set the state of all power switches simultaneously. If the `device` argument is omitted, it controls only one Octofet board.
+
+### `void digitalPortWrite(uint8_t value, uint8_t device = 0)`
+
+Sets the state ("on" or "off") for all power switches. Function arguments:
+
+- `value`: 8-bit value for all 8 power switches. One bit for each power switch: `1` is "on" and `0` is "off".
+- `device`: the ordinal number of the device on the bus. It ranges from `0` to `n - 1`, where `n` is the last device number. Use an optional value `ALL` to select and set the state of all devices. If the `device` argument is omitted, it controls only one Octofet board.
+
+### `bool digitalPinRead(uint8_t pin, uint8_t device = 0)`
+
+Reads the state ("on" or "off") of a single power switch of the device. Returns `true` or `false`. Function arguments:
+
+- `pin`: the ordinal number power switch on the device. It ranges from `0` to `7`.
+- `device`: the ordinal number of the device on the bus. It ranges from `0` to `n - 1`, where `n` is the last device number. If the `device` argument is omitted, it controls only one Octofet board.
+
+### `uint8_t digitalPortRead(uint8_t device = 0)`
+
+Reads state (turn-on or turn-off) of all power switches of the device. Returns the 8-bit value of all 8 power switches. One bit for each power switch: `1` is "on" and `0` is "off". Function argument `device`: the ordinal number of the device on the bus. It ranges from `0` to `n - 1`, where `n` is the last number device. If the `device` argument is omitted, it controls only one Octofet board.
+
+### `void reset()`
+
+Resets values of all power switches of all devices to the default state.
+
+- If the `resetValues` argument in the `Octofet` constructor is adjusted, power switches are set by the `resetValues`.
+- If the `resetValues` argument is omitted, power switches are set to the "off" state.
```

Comment:
Why the `reset` concept is even necessary? I see how it makes things more complicated: more concepts, more arguments, more hidden state. And see no benefits.

Reply:
Not all reset states are all zeros. For example "smart home" after reset:
The main light is off.
The standby light is on.
Heaters are off.
The refrigerator is on.
The lock is close. Etc.
The user will want to select the necessary state configuration for him after a reset. Yes, it can solve the problem using hardware "state inverters", but these devices increase complexity and reduce reliability.

Reply:
Still does not make sense to me. Does the Octofet module stores defaults across reboots and power on/offs? If not, how this:

```cpp
uint8_t state = 0b10101010;
octofet.begin(&state);
```

is better than the following?

```cpp
octofet.begin();
octofet.digitalWrite8(0b10101010);
```

Reply:
1. Reset happens not only at the beginning of the program.
2. If state before is 0b1001 where the first bit is inductive device or device with a long reboot time then  0b1001 -> 0b0001 is more better then 0b1001 -> 0b0000 -> 0b0001.